0009-simple-payment-setup-protocol/0009-simple-payment-setup-protocol.md

```diff
@@ -62,6 +63,17 @@ Host: example.com
 Accept: application/spsp4+json, application/spsp+json
 ```
 
+##### Request Headers
+
+The request MAY contain at least the following headers in order to pre-share STREAM Receipt details between the SPSP Server and a party verifying payments sent to the SPSP Server:
+
+| Header          | Description                                                |
+|:----------------|:-----------------------------------------------------------|
+| `Receipt-Nonce`  | A base64-encoded 16-byte random nonce used to identify the STREAM connection in STREAM Receipts. |
+| `Receipt-Secret` | A base64-encoded 32-byte key used to generate a STREAM Receipt's HMAC. |
+
+The SPSP Client MAY be provided with an SPSP Endpoint belonging to the receipt verifier, which would add the receipt headers and proxy the query to the SPSP Server.
```

Comment:
How does the verifier know from the response whether the SPSP server supports receipts or not?

Reply:
:+1: we should probably add some kind of discovery for this, either in a response header or in the response body for SPSP

Reply:
Are we thinking the verifier would return a 4XX error if the SPSP server doesn't support receipts?
If so, maybe the SPSP server could just parrot the Receipt-Nonce header if it supports receipts.

Reply:
We could include `"receipts": true` in the SPSP response body. I feel like that's a little more clean than echoing the header. Another option would be some kind of SPSP features header if we wanted to make it extensible

Reply:
added optional `receipts` field to SPSP response body: https://github.com/interledger/rfcs/pull/570/commits/1633e86180f268e4808f9011610f91b6f325096b